### PR TITLE
feat: [CDS-49450]:  added ASG Rolling Rollback step

### PR DESCRIPTION
**Pull request by vikrant-harness**
#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/75-cd/components/PipelineSteps/AsgRollingRollbackStep/AsgRollingRollbackStep.tsx [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+import { VariablesListTable } from '@pipeline/components/VariablesListTable/VariablesListTable'
+import { StepType } from '@pipeline/components/PipelineSteps/PipelineStepInterface'
+import { PipelineStep } from '@pipeline/components/PipelineSteps/PipelineStep'
+import { GenericExecutionStepEditRef } from '../Common/GenericExecutionStep/GenericExecutionStepEdit'
+import { GenericExecutionStepInputSet } from '../Common/GenericExecutionStep/GenericExecutionStepInputSet'
+import { validateGenericFields } from '../Common/GenericExecutionStep/utils'
+import pipelineVariableCss from '@pipeline/components/PipelineStudio/PipelineVariables/PipelineVariables.module.scss'
+
+interface AsgRollingRollbackStepVariableStepProps {
+  initialValues: StepElementConfig
```
Comment (pankajsingh-harness) [inline src/modules/75-cd/components/PipelineSteps/AsgRollingRollbackStep/AsgRollingRollbackStep.tsx:26]: `initialValues: StepElementConfig
  stageIdentifier: string
  onUpdate?(data: StepElementConfig): void`
  
  These fields are not used from this interface. So can we remove them as of now. We can add them later when needed.

File: src/modules/75-cd/components/PipelineSteps/AsgRollingRollbackStep/__tests__/AsgRollingRollback.test.tsx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+const existingInitialValues = {
+  identifier: 'Step_AsgRollingRollback',
+  name: 'Step AsgRollingRollback',
+  timeout: '10m',
+  type: StepType.AsgRollingRollback
+}
+const onUpdate = jest.fn()
+const onChange = jest.fn()
+
+describe('Asg Canary Delete Step tests', () => {
```
Comment (chetanvekariya-harness) [inline src/modules/75-cd/components/PipelineSteps/AsgRollingRollbackStep/__tests__/AsgRollingRollback.test.tsx:30]: This title needs a change, I guess.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/75-cd/components/PipelineSteps/AsgRollingRollbackStep/__tests__/AsgRollingRollback.test.tsx (hunk 2 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/AsgRollingRollbackStep/__tests__/AsgRollingRollback.test.tsx:30]: Done

File: src/modules/75-cd/components/PipelineSteps/AsgRollingRollbackStep/AsgRollingRollbackStep.tsx (hunk 1 above)
Comment (vikrant-harness) [reply src/modules/75-cd/components/PipelineSteps/AsgRollingRollbackStep/AsgRollingRollbackStep.tsx:26]: Done

**Review by pankajsingh-harness — DISMISSED** [review]

**Review by chetanvekariya-harness — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Pull request merged**